examples/index.js

```diff
@@ -0,0 +1,41 @@
+const Fastify = require('fastify')
+
+async function run({ token, ...options }) {
+  if (!token) {
+    throw new Error('Please provide a token via --token {token}')
+  }
+
+  const fastify = Fastify()
+
+  await fastify.register(require('..'), options)
+
+  fastify.get(
+    '/',
+    {
+      preValidation: fastify.authenticate
+    },
+    r => r.user
+  )
+
+  await fastify.listen()
+
+  const response = await fastify.inject({
+    url: '/',
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  })
+
+  console.log('\n==============\n|  Response  |\n==============\n')
+  console.log(JSON.stringify(JSON.parse(response.body), null, 2))
+
+  fastify.close()
+}
+
+run(
+  process.argv
+    .slice(2)
+    .map((o, i, _) => !(i % 2) && [o, _[i + 1]])
```

Comment:
This is at the same amazing and terrifing.
For better readability, shall we use something like minimist?

Reply:
I didn't want to add a dependency for the sake of a simple example. This is not code we're releasing

Reply:
As discussed, I'll keep this to avoid introducing a dependency